vendor/engines/bulk_email/spec/controllers/bulk_email/bulk_email_controller_spec.rb

```diff
@@ -112,19 +112,31 @@
       end
     end
 
-    context "when rendering as csv" do
-      let(:user_types) { %i(customers) }
+  end
 
-      before(:each) do
-        maybe_grant_always_sign_in :director
-        @params[:format] = "csv"
-        do_request
-      end
+  describe "POST #download" do
+    let(:recipients) { FactoryGirl.create_list(:user, 3) }
+    let(:expected_csv_content) { csv_header + "\n" + expected_csv_body + "\n" }
+    let(:csv_header) { "Name,Username,Email" }
+    let(:expected_csv_body) do
+      recipients.map do |user|
+        [user.full_name, user.username, user.email].join(",")
+      end.join("\n")
+    end
 
-      it "sets the filename" do
-        expect(response.headers["Content-Disposition"])
-          .to eq('attachment; filename="bulk_email_customers.csv"')
-      end
+    before(:each) do
+      @action = "download"
+      @method = :post
+      @params.merge!(format: :csv, recipient_ids: recipients.map(&:id))
```

Comment:
I don't really care about the 

> Use `@params[:format] = :csv; @params[:recipient_ids] = recipients.map(&:id)` instead of `@params.merge!(format: :csv, recipient_ids: recipients.map(&:id))` cop. 

I'd be fine if we disabled that one.